pkg/scalers/rabbitmq_scaler.go

```diff
@@ -99,11 +99,15 @@ func parseRabbitMQMetadata(resolvedEnv, metadata, authParams map[string]string)
 			return nil, fmt.Errorf("no apiHost setting given")
 		}
 	} else {
-		if val, ok := authParams["host"]; ok {
+		if val, ok := metadata["localhost"]; ok {
+			hostSetting := val
+			if val, ok := resolvedEnv[hostSetting]; ok {
+				meta.host = val
+			}
```

Comment:
I'm not sure I understand what this code is doing. It seems to be doing exactly the same thing as the `if val, ok := metadata["host"]; ok` below, right?

Reply:
Yes. but it reference the `localhost` one. This code take the host url from the enviornment variable on the target pod. 
`metadata["localhost"]` returns the value of `LocalHost` that is defined on the secret. `resolvedEnv[hostSettings]` get the value of the enviornment variables `LocalHost` that is injected by the `envFrom`. 

So that, `if val, ok:= metadata["host"]; ok` val will be the `amqp://user:PASSWORD@rabbitmq.default.svc.cluster.local:5672` `if val, ok := metadata["localhost"]; ok` val will be the `ampqb://user:PASSWORD@localhost:5672` that is decoded value of 
`YW1xcDovL3VzZXI6UEFTU1dPUkRAMTI3LjAuMC4xOjU2NzI=` That means, local running keda on your PC can access to the k8s cluster through `kubectl port-forward service/rabbitmq 5672`.